### DISCUSSION TITLE
Any function to move the move/rotate/scale widgets to the origin of a model?

### DISCUSSION THREAD

**Discussion by Hipshot-zz** [Garux/netradiant-custom#217] [General]
Like, I assume it places these widgets in the middle of the bounding box now, but it would be a lot more useful if they are set at the origin of models. I understand that this behaviour won't work with brushes of course.

**Comment by Garux** [comment]
This was intentional highly requested change to use bbox of models. Bbox way provides many means to edit easily, which were impossible otherwise.
There is no way to put transform origin to model origin automatically now, only can move it there manually.
In which case you consider using origin as being a lot more useful?

**Comment by Hipshot-zz** [comment]
I would love to be able to snap that to the origin of a model, either as an permanent option or maybe a button or function that snaps it there. When you use models a bit more, it starts to get annoying when you especially rotates. So, to answer, for me it would be very useful.

**Reply by Garux** [reply]
I ask *in which concrete usecase* you consider using origin as being a lot more useful.
From our experience using origin, which has no guaranteed relation to model geometry, is clearly nonsense and does not worth GUI and code complication.
Knowing exact usecase may help to address the problem in most optimal way.

**Reply by Garux** [reply]
Probably this new feature can serve when model is aligned via rotation.

![BBox rotate-snap (m1 + alt + ctrl), move along plane (m1 + ctrl + shit)](https://garux.github.io/NRC/gifs/35rotateSnap_planeMove.gif)
https://garux.github.io/NRC/#BBox%20rotate-snap%20(m1%20+%20alt%20+%20ctrl),%20move%20along%20plane%20(m1%20+%20ctrl%20+%20shift)
